python/tests/test_nvcomp.py

```diff
@@ -10,60 +10,458 @@
+    compressor = libnvcomp.CascadedManager(**inputs)
+    final = compressor.compress(data)
+    assert len(final) == 600
+
+
+@pytest.mark.parametrize(
+    "inputs",
+    [
+        {},
+        {"chunk_size": 1 << 16, "algo": 0, "device_id": 0},
+        {
+            "chunk_size": 1 << 16,
+        },
+        {
+            "algo": 0,
+        },
+        {
+            "device_id": 0,
+        },
+    ],
+)
+def test_gdeflate_inputs(inputs):
+    size = 10000
+    dtype = inputs.get("data_type") if inputs.get("data_type") else np.int8
+    data = cupy.array(np.arange(0, size // dtype(0).itemsize, dtype=dtype))
+    compressor = libnvcomp.GdeflateManager(**inputs)
+    final = compressor.compress(data)
+    assert len(final) == 732
+
+
+@pytest.mark.parametrize(
+    "inputs, expected",
+    zip(
+        [
+            {"algo": 0},
+        ],
+        [732],
+    ),
+)
+def test_gdeflate_algorithms(inputs, expected):
+    size = 10000
+    dtype = np.int8
+    data = cupy.array(np.arange(0, size // dtype(0).itemsize, dtype=dtype))
+    compressor = libnvcomp.GdeflateManager(**inputs)
+    final = compressor.compress(data)
+    assert len(final) == expected
+
+
+@pytest.mark.xfail(raises=ValueError)
+@pytest.mark.parametrize(
+    "inputs, expected", zip([{"algo": 1}, {"algo": 2}], [732, 732])
+)
+def test_gdeflate_algorithms_not_implemented(inputs, expected):
+    size = 10000
+    dtype = np.int8
+    data = cupy.array(np.arange(0, size // dtype(0).itemsize, dtype=dtype))
+    compressor = libnvcomp.GdeflateManager(**inputs)
+    final = compressor.compress(data)
+    assert len(final) == expected
+
+
+@pytest.mark.parametrize(
+    "inputs",
+    [
+        {},
+        {"chunk_size": 1 << 16, "data_type": np.uint8, "device_id": 0},
+        {
+            "chunk_size": 1 << 16,
+        },
+        {
+            "data_type": np.uint8,
+        },
+        {
+            "device_id": 0,
+        },
+    ],
+)
+def test_lz4_inputs(inputs):
+    size = 10000
+    dtype = inputs.get("data_type") if inputs.get("data_type") else np.int8
+    data = cupy.array(np.arange(0, size // dtype(0).itemsize, dtype=dtype))
+    compressor = libnvcomp.LZ4Manager(**inputs)
+    final = compressor.compress(data)
+    assert len(final) == 393
+
+
+@pytest.mark.parametrize(
+    "inputs",
+    [
+        {},
+        {"chunk_size": 1 << 16, "device_id": 0},
+        {
+            "chunk_size": 1 << 16,
+        },
+        {"device_id": 0},
+    ],
+)
+def test_snappy_inputs(inputs):
+    size = 10000
+    dtype = np.int8
+    data = cupy.array(np.arange(0, size // dtype(0).itemsize, dtype=dtype))
+    compressor = libnvcomp.SnappyManager(**inputs)
+    final = compressor.compress(data)
+    assert len(final) == 3548
+
+
+@pytest.mark.parametrize(
+    "compressor_size",
+    zip(
+        [
+            libnvcomp.LZ4Manager,
+            libnvcomp.CascadedManager,
+            libnvcomp.SnappyManager,
```

Comment:
Did we want to support the full set of managers here?

Reply:
Thanks for reminding me, hitting that, then merging.